api/hasura/event-triggers/create-profile.ts

```diff
@@ -0,0 +1,54 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+
+import { gql } from '../../../api-lib/Gql';
+import { EventTriggerPayload } from '../../../api-lib/types';
+import { verifyHasuraRequestMiddleware } from '../../../api-lib/validate';
+import { profiles_constraint } from '../../../src/lib/gql/zeusHasuraAdmin';
+
+async function handler(req: VercelRequest, res: VercelResponse) {
+  const {
+    event: { data },
+  }: EventTriggerPayload<'users', 'UPDATE' | 'INSERT'> = req.body;
+
+  try {
+    if (data.old?.address !== data.new.address) {
+      const { insert_profiles_one } = await gql.q('mutation')({
+        // Create a profile if none exists yet
+        insert_profiles_one: [
+          {
+            object: { address: data.new.address },
+
+            // This clause allows gql to catch the conflict and do nothing
+            // hasura calls this an "upsert"
+            on_conflict: {
+              constraint: profiles_constraint.profiles_address_key,
+              // Don't update the entry at all if a profile exists
+              // Don't want to touch the timestamp if we aren't actually
+              // modifying anything
+              update_columns: [],
+            },
+          },
+          {
+            address: true,
+          },
+        ],
+      });
+      if (insert_profiles_one) {
+        res.status(200).json({
+          message: `profile created for ${insert_profiles_one.address}`,
+        });
+        return;
+      }
+      res
```

Comment:
I'm new to gql - is this how you know an insert had a pk constraint violation? it just returns undefined (rather than throw) ?

Reply:
@CryptoGraffe no this is just logging for Hasura. The config of this call "swallows" any conflict errors. See here for more details: 

https://hasura.io/docs/latest/graphql/core/databases/postgres/mutations/upsert.html